src/Compilers/CSharp/Portable/Errors/LazyObsoleteDiagnosticInfo.cs

```diff
@@ -33,10 +33,10 @@ internal override DiagnosticInfo GetResolvedInfo()
 
                 if (_symbol.ObsoleteState == ThreeState.True)
                 {
-                    var inObsoleteContext = ObsoleteAttributeHelpers.GetObsoleteContextState(_containingSymbol, forceComplete: true);
-                    Debug.Assert(inObsoleteContext != ThreeState.Unknown);
+                    var kind = _containingSymbol.GetObsoleteContextKind(forceComplete: true);
+                    Debug.Assert(kind != ObsoleteAttributeKind.Uninitialized);
 
-                    if (inObsoleteContext == ThreeState.False)
+                    if (kind == ObsoleteAttributeKind.None)
```

Comment:
Yes, good catch. Working on a test that hits this code path for an Experimental symbol in an Obsolete context.